qiskit_experiments/analysis/curve_fitting.py

```diff
@@ -0,0 +1,291 @@
+
+    Returns:
+        result containing `popt` the optimal fit parameters,
+        `popt_err` the standard error estimates popt,
+        `pcov` the covariance matrix for the fit,
+        `chisq` the chi-squared parameter of fit,
+        `xrange` the range of xdata values used for fit.
+
+    Raises:
+        QiskitError: if the number of y-values in the dataset is not
+                     greater than the number of parameters in the
+                     fit function.
+    """
+    # Format p0 parameters if specified as dictionary
+    if isinstance(p0, dict):
+        param_keys = list(p0.keys())
+        param_p0 = list(p0.values())
+
+        # Convert bounds
+        if bounds:
+            lower = [bounds[key][0] for key in param_keys]
+            upper = [bounds[key][1] for key in param_keys]
+            param_bounds = (lower, upper)
+        else:
+            param_bounds = None
+
+        # Convert fit function
+        def fit_func(x, *params):
+            return func(x, **dict(zip(param_keys, params)))
+
+    else:
+        param_keys = None
+        param_p0 = p0
+        param_bounds = bounds
+        fit_func = func
+
+    # Check data set has at least 1 more values than model parameters
+    num_params = len(param_p0)
+    if len(ydata) <= num_params:
+        raise QiskitError(
+            "The number of y-values in the dataset is not greater than"
+            " the number of parameters in the fit function."
+        )
+
+    # Override scipy.curve_fit default for absolute_sigma to True
+    # if not specified in kwargs
+    if sigma is not None and "absolute_sigma" not in kwargs:
+        kwargs["absolute_sigma"] = True
+
+    # Run curve fit
+    # TODO: Add error handling so if fitting fails we can return an analysis
+    #       result containing this information
+    # pylint: disable = unbalanced-tuple-unpacking
+    popt, pcov = opt.curve_fit(
+        fit_func, xdata, ydata, sigma=sigma, p0=param_p0, bounds=param_bounds, **kwargs
+    )
+    popt_err = np.sqrt(np.diag(pcov))
+
+    # Calculate the reduced chi-squared for fit
+    yfits = fit_func(xdata, *popt)
+    residues = (yfits - ydata) ** 2
+    if sigma is not None:
+        residues = residues / (sigma ** 2)
+    chisq = np.sum(residues) / (len(yfits) - num_params)
+
+    # Compute xdata range for fit
+    xdata_range = [min(xdata), max(xdata)]
+
+    result = {
+        "popt": popt,
+        "popt_keys": param_keys,
+        "popt_err": popt_err,
+        "pcov": pcov,
+        "chisq": chisq,
+        "xrange": xdata_range,
+    }
+
+    return AnalysisResult(result)
+
+
+def multi_curve_fit(
+    funcs: List[Callable],
+    series: np.ndarray,
+    xdata: np.ndarray,
+    ydata: np.ndarray,
+    p0: np.ndarray,
+    sigma: Optional[np.ndarray] = None,
+    weights: Optional[np.ndarray] = None,
+    bounds: Optional[Union[Dict[str, Tuple[float, float]], Tuple[np.ndarray, np.ndarray]]] = None,
+    **kwargs,
+):
```

Comment:
Check if the comments about `sigma` and `absolute_sigma` are relevant also for `multi_curve_fit`